com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java

```diff
@@ -474,27 +511,40 @@ private ParserRuleContext parsePropertyValue(LuaParser.ArgsContext argsCtx, Prop
                     }
                 }
                 else {
-                    String firstStrArg = getFirstStringArg(ctx.nameAndArgs().args());
                     if (fnDesc.isObject("resource")) {
                         property.type = PropertyType.PROPERTY_TYPE_HASH;
                         property.isResource = true;
                         resultContext = ctx;
+                        String firstStrArg = getFirstStringArg(ctx.nameAndArgs().args());
+                        property.value = firstStrArg == null ? "" : firstStrArg;
                     }
                     else if (fnDesc.is("hash", null) || fnDesc.is("hash", "_G")) {
                         property.type = PropertyType.PROPERTY_TYPE_HASH;
+                        String firstStrArg = getFirstStringArg(ctx.nameAndArgs().args());
                         // hash(arg) requires an argument
                         if (firstStrArg != null) {
                             resultContext = ctx;
                         }
+                        property.value = firstStrArg == null ? "" : firstStrArg;
                     }
                     else if (fnDesc.is("url", "msg")) {
                         property.type = PropertyType.PROPERTY_TYPE_URL;
                         resultContext = ctx;
+                        List<String> allArgs = getAllStringArgs(ctx.nameAndArgs().args());
+                        property.value = concatenateUrl(allArgs);
                     }
-                    property.value = firstStrArg == null ? "" : firstStrArg;
                 }
             }
         }
         return resultContext;
     }
+    public static String concatenateUrl(List<String> allArgs) {
```

Comment:
Url concatinator - easy way to create an url from a list of strings

Reply:
Does this handle, size == 2 ?

Reply:
Fixed, thank you!